docs/src/interface.md

```diff
@@ -0,0 +1,14 @@
+# Interface
+
+This is the public interface that day-to-day users of AD are expected to interact with if
+for some reason DifferentiationInterface.jl does not suffice.
+If you have not tried using Mooncake.jl via DifferentiationInterface.jl, please do so.
+See [Tutorial](@ref) for more info.
+
+```@docs; canonical=false
```

Comment:
```suggestion
```@docs; canonical=true
```
There needs to be at least one canonical location for the docstring, otherwise I think references won't work properly.
This also applies to private symbols by the way. My suggestion to put `canonical = false` for docstrings inside text was based on the assumption that there would also be a "list of private docstrings" somewhere, mirroring the public API, with `canonical = true`. Does that make sense?

Reply:
Right you are. I had to apply a filter to handle the unexported bits of the public interface, but I think I've now got it working.

Reply:
Yeah, I meant to show you, I had to do the same in DI

https://github.com/JuliaDiff/DifferentiationInterface.jl/blob/58605dc2c91922c20cbefd24197bd6cd809a2383/DifferentiationInterface/docs/src/api.md?plain=1#L134-L142